Cabal/src/Distribution/Simple/GHC.hs

```diff
@@ -1207,7 +1209,8 @@ gbuildSources verbosity specVer tmpDir bm =
       let mainModName = fromMaybe ModuleName.main $ exeMainModuleName exe
           otherModNames = exeModules exe
 
-      if isHaskell main
+      -- Scripts have fakePackageId and are always Haskell but can have any extension.
+      if isHaskell main || pkgId == fakePackageId
```

Comment:
the code is using fakePackageId to check if it is handling a script, but the fake package has other uses (cabal install iirc), could it interfere in some way?

Reply:
The only use of `fakePackageId` other than scripts is for the global (outside of a project context) repl, in which case there is no exe to be built. The only case where there is both an exe that is being built and `fakePackageId` is set is if we are running a script.

evidence: `fakePackageId` is only set in StriptUtil.hs in the function `fakeProjectSourcePackage`. `fakeProjectSourcePackage` is only used internally and in CmdRepl.hs